store/sqlstore/group_store.go

```diff
@@ -909,6 +909,41 @@ func (s *SqlGroupStore) groupsBySyncableBaseQuery(st model.GroupSyncableType, t
 			OrderBy("ug.DisplayName")
 	}
 
+	if opts.FilterAllowReference && t == selectGroups {
+		query = query.Where("ug.AllowReference = 1")
+	}
+
+	if len(opts.Q) > 0 {
+		pattern := fmt.Sprintf("%%%s%%", sanitizeSearchTerm(opts.Q, "\\"))
+		operatorKeyword := "ILIKE"
+		if s.DriverName() == model.DATABASE_DRIVER_MYSQL {
+			operatorKeyword = "LIKE"
+		}
+		query = query.Where(fmt.Sprintf("(ug.Name %[1]s ? OR ug.DisplayName %[1]s ?)", operatorKeyword), pattern, pattern)
+	}
+
+	return query
+}
+
+func (s *SqlGroupStore) getGroupsAssociatedToChannelsByTeam(st model.GroupSyncableType, teamID string, opts model.GroupSearchOpts) sq.SelectBuilder {
+	query := s.getQueryBuilder().
+		Select(fmt.Sprintf("gc.ChannelId, ug.*, gc.SchemeAdmin AS SyncableSchemeAdmin")).
+		From("UserGroups ug").
+		LeftJoin(fmt.Sprintf("(SELECT GroupChannels.GroupId, GroupChannels.ChannelId, GroupChannels.DeleteAt, GroupChannels.SchemeAdmin FROM GroupChannels LEFT JOIN Channels ON Channels.TeamId = ? WHERE GroupChannels.DeleteAt = 0 AND Channels.DeleteAt = 0 AND GroupChannels.ChannelId = Channels.Id) AS gc ON gc.GroupId = ug.Id"), teamID).Where(fmt.Sprintf("ug.DeleteAt = 0 AND gc.DeleteAt = 0")).OrderBy("ug.DisplayName")
```

Comment:
I think breaking up this into multiple lines - something like this
```suggestion
		LeftJoin(fmt.Sprintf("
		(SELECT 
		    GroupChannels.GroupId, GroupChannels.ChannelId, GroupChannels.DeleteAt, GroupChannels.SchemeAdmin 
		FROM 
		    GroupChannels 
		LEFT JOIN 
		    Channels ON Channels.TeamId = ? 
		WHERE 
		    GroupChannels.DeleteAt = 0 
		    AND Channels.DeleteAt = 0 
		    AND GroupChannels.ChannelId = Channels.Id
	        ) AS gc ON gc.GroupId = ug.Id"), teamID)
	        .Where("ug.DeleteAt = 0 AND gc.DeleteAt = 0")
	        .OrderBy("ug.DisplayName")
```

would make it a lot more readable